### PR TITLE
Fix implementors display

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Part of https://github.com/rust-lang/rust/issues/86632.

This PR does a few things:
 * It fixes of the JS rendered implementors.
 * It generates anchors for JS rendered implementors to make it coherent with the others.
 * It adds a test to ensure that we won't have the same issue again.
 * It changes the way we render the rustdoc-gui crates to simplify it a bit and also to allow to have dependencies without going through compiletest.

Before:

Screenshot from 2021-07-10 13-30-13

After:

Screenshot from 2021-07-10 21-11-15

I plan to add the `[src]` links in another PR because this one is already big enough.

cc @Mark-Simulacrum (for the bootstrap changes)

r? @Nemo157

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
cc @camelid

**Review by notriddle — APPROVED** [review]

File: src/bootstrap/test.rs [hunk 1]
```diff
@@ -907,27 +907,26 @@ impl Step for RustdocGUI {
         // We remove existing folder to be sure there won't be artifacts remaining.
         let _ = fs::remove_dir_all(&out_dir);
 
-        let mut nb_generated = 0;
+        // let mut nb_generated = 0;
```
Comment (notriddle) [inline src/bootstrap/test.rs:910]: Looks like you forgot to remove this?

Comment (GuillaumeGomez) [reply src/bootstrap/test.rs:910]: Indeed, good catch!

**Pull request merged**